src/Features/CSharp/Portable/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs

```diff
@@ -274,16 +278,29 @@ where outSymbol.Equals(symbol)
 
             // Ensure we can track the references and the local variable as we make edits
             // to the tree.
-            var rootWithTrackedNodes = root.TrackNodes(references.Concat(localDeclarator).Concat(enclosingBlock));
+            var rootWithTrackedNodes = root.TrackNodes(
+                references.Concat(ImmutableArray.Create<SyntaxNode>(localDeclarator, localDeclaration, enclosingBlock)));
 
             // Now, take the local variable and remove it's initializer.  Then go to all
             // the locations where we read from it.  If they're definitely assigned, then
             // that means the out-var did it's work and assigned the variable across all
             // paths. If it's not definitely assigned, then we can't inline this variable.
             var currentLocalDeclarator = rootWithTrackedNodes.GetCurrentNode(localDeclarator);
+            var currentLocalDeclaration = rootWithTrackedNodes.GetCurrentNode(localDeclaration);
+            var updatedDeclaration = currentLocalDeclaration
+                .ReplaceNode(currentLocalDeclarator, currentLocalDeclarator.WithInitializer(null));
+
+            // If the declaration was a "var" declaration, then replace "var" with the actual
+            // type of the local.  This way we don't get a "'var v' requires an initializer" which
+            // will suppress the message about definite assignment later.
+            if (updatedDeclaration.Type.IsVar)
+            {
+                updatedDeclaration = updatedDeclaration.WithType(
+                    outLocalSymbol.Type.GenerateTypeSyntax());
+            }
+
             var rootWithoutInitializer = rootWithTrackedNodes.ReplaceNode(
-                currentLocalDeclarator,
-                currentLocalDeclarator.WithInitializer(null));
```

Comment:
It was my mistake. I was picturing `WithInitializer(null)` producing a `null` value expression, like this:

```csharp
var item = null;
```

In reality, this call is removing the initializer expression altogether, like this:

```csharp
var item;
```

My question, which is now clear does not apply, was asking if you could do this:

```csharp
var item = default(SomeType);
```